### PR TITLE
Cleanups to glass files and libGenIC globals

### PR DESCRIPTION
**Pull request by sbird**
This cleans up our glass file support - MakeGlass is replaced with two options, MakeGlassCDM and MakeGlassBar for baryon and CDM glass file support respectively. Neutrinos can't be put into a glass (it makes little sense). The GenIC global particle table variable is removed in preparation (for something I am about to try, but it is a cleanup that makes sense on its own).

### CODE REVIEW

File: genic/main.c [hunk 1]
```diff
@@ -123,32 +127,30 @@ int main(int argc, char **argv)
...
-  write_particle_data(1, &bf, 0, All2.Ngrid);
-  free_ffts();
+  write_particle_data(1, &bf, 0, All2.Ngrid, ICP, NumPart);
 
   /*Now make the gas if required*/
   if(All2.ProduceGas) {
 
-    if(!All2.MakeGlass) {
-        setup_grid(shift_gas, All2.Ngrid);
+    if(!All2.MakeGlassBar) {
```
Comment (rainwoodman) [inline genic/main.c]: You sure want to go with Bar not Gas here?

Comment (sbird) [reply genic/main.c]: You're right! I want to go with Gas

File: libgenic/glass.c [hunk 2]
```diff
@@ -91,7 +69,13 @@ setup_glass(double shift, int Ngrid, int seed)
 
     gsl_rng_free(rng);
 
+    evolve_glass(seed, ICP, NumPart);
+    return NumPart;
+}
 
+void evolve_glass(int seed, struct ic_part_data * ICP, const int NumPart)
```
Comment (rainwoodman) [inline libgenic/glass.c]: Perhaps mark this as static and remove it from the .h file, since it is not used outside this module.

Comment (sbird) [reply libgenic/glass.c]: I have a feature I'm trying out that uses it in main.c If it doesn't turn out to be useful I'll remove this from the header

**Comment by rainwoodman** [comment]
Looks good. And minor suggestions -- Why Bar not Gas?

File: libgenic/glass.c (hunk 2 above)
Comment (rainwoodman) [reply libgenic/glass.c]: If we are exposing more functions, then the functions in the module shall be properly prefixed ..

**Comment by sbird** [comment]
Fixed! Thanks! Will merge when green.

File: libgenic/glass.c (hunk 2 above)
Comment (sbird) [reply libgenic/glass.c]: glass_evolve ?

Comment (sbird) [reply libgenic/glass.c]: Prefixed and some comments added

**Pull request merged**